Editor/Window/Utils/StateManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+
+namespace Editor.Window.Utils
+{
+    public class StateManager
+    {
+        private State _state;
+        public CoreApi CoreApi { get; set; }
+        public GameLiftRequestAdapter RequestAdapter { get; set; }
+        public State State { get; set; }
+        public IAmazonGameLiftClientWrapper GameLiftWrapper { get; private set; }
+        
+        public IAmazonGameLiftClientFactory AmazonGameLiftClientFactory { get; set; }
+        
+        public void SetupClientFactory()
+        {
+            AmazonGameLiftClientFactory = new AmazonGameLiftClientFactory(CoreApi);
+        }
+        
+        public void SetupWrapper()
+        {
+            GameLiftWrapper = AmazonGameLiftClientFactory.Get(_state.SelectedProfile);
+        }
+
+        public void SetupRequestAdapter()
+        {
+            RequestAdapter = new GameLiftRequestAdapter(CoreApi, (AmazonGameLiftWrapper)GameLiftWrapper);
+        }
```

Comment:
These two methods should be handled by a single method, `SetProfile`, instead of allowing anyone to modify the state object and expecting them to call the correct setup methods afterward.

Reply:
Merged